src/Datadog.Trace/Propagation/W3CSpanContextPropagator.cs

```diff
@@ -0,0 +1,94 @@
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using Datadog.Trace.Conventions;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.Propagation
+{
+    internal class W3CSpanContextPropagator : IPropagator
+    {
+        private const string TraceParentFormat = "00-{0}-{1}-01";
+
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor<W3CSpanContextPropagator>();
+        private static readonly int VersionPrefixIdLength = "00-".Length;
+        private static readonly int TraceIdLength = "0af7651916cd43dd8448eb211c80319c".Length;
+        private static readonly int VersionAndTraceIdLength = "00-0af7651916cd43dd8448eb211c80319c-".Length;
+        private static readonly int SpanIdLength = "00f067aa0ba902b7".Length;
+
+        private readonly ITraceIdConvention _traceIdConvention;
+
+        public W3CSpanContextPropagator(ITraceIdConvention traceIdConvention)
+        {
+            _traceIdConvention = traceIdConvention;
+        }
+
+        public void Inject<T>(SpanContext context, T carrier, Action<T, string, string> setter)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            if (carrier == null)
+            {
+                throw new ArgumentNullException(nameof(carrier));
+            }
+
+            if (setter == null)
+            {
+                throw new ArgumentNullException(nameof(setter));
+            }
```

Comment:
can you "collate" it into 3 lines, please 😄

Reply:
I know this is just a matter of styling, but I hate one line ifs. I would rather leave it as it is if we don't have strict rules on such things.

Reply:
TBH I see no value for null checks for internals :-)

Reply:
Yep, I did it to be consistent with other propagators. TBH if nobody opposes I would love to get rid of that :)